Sources/ConsoleKit/Command/Commands.swift

```diff
@@ -45,12 +50,24 @@ public struct Commands {
     ///     - help: Optional help messages to include.
     /// - returns: A `CommandGroup` with commands and defaultCommand configured.
     public func group(help: String = "") -> CommandGroup {
-        return _Group(commands: self.commands, defaultCommand: self.defaultCommand, help: help)
+        var group = _Group(
+            commands: self.commands,
+            defaultCommand: self.defaultCommand,
+            help: help
+        )
+        if self.enableAutocomplete {
+            // First a placeholder uninitialized autocomplete command is added to the commands. The
+            // second, _initialized_ autocomplete command immediately overwrites the first, but will
+            // use it to provide completion for itself!
+            group.commands["autocomplete"] = GenerateAutocompleteCommand()
+            group.commands["autocomplete"] = GenerateAutocompleteCommand(rootCommand: group)
+        }
+        return group
```

Comment:
This is the tricky dance that I thought best to keep as an internal implementation detail.